File: done/kyu-8/sum-of-the-first-nth-term-of-series.php
<?php

// https://www.codewars.com/kata/sum-of-the-first-nth-term-of-series/train/php

function series_sum(int $n): float
{
    $s = 0;
    for ($i = 1; $i <= $n; $i++) { 
        $s += 1 / (($i - 1) * 3 + 1);
    }

    return round($s, 2);
}

File: Elixir/Kyu7/sum-of-the-first-nth-term-of-series.ex
# https://www.codewars.com/kata/sum-of-the-first-nth-term-of-series

defmodule Series do
  def sum(0), do: "0.00"
  def sum(1), do: "1.00"

  def sum(n) do
    2..n
    |> Enum.to_list()
    |> Enum.map(fn x -> member(x) end)
    |> Enum.sum()
    |> Kernel.+(1)
    |> Float.round(2)
    |> to_string()
  end

  defp member(index), do: 1 / ((index - 1) * 3 + 1)
end
